src/client.js

```diff
@@ -286,34 +285,40 @@ Client.prototype.connect = function(args) {
 	client.save();
 };
 
-Client.prototype.updateToken = function(callback) {
-	var client = this;
-
-	crypto.randomBytes(48, function(err, buf) {
+Client.prototype.generateToken = function(callback) {
+	crypto.randomBytes(48, (err, buf) => {
 		if (err) {
 			throw err;
 		}
 
-		callback(client.config.token = buf.toString("hex"));
+		callback(buf.toString("hex"));
 	});
 };
 
+Client.prototype.updateSession = function(token, ip, request) {
+	const client = this;
+	const agent = UAParser(request.headers["user-agent"] || "");
+
+	client.config.sessions[token] = {
+		lastUse: Date.now(),
+		ip: ip,
+		agent: `${agent.browser.name} ${agent.browser.major} on ${agent.os.name} ${agent.os.version}`,
```

Comment:
Ah, that makes sense I guess. Can you then use the format I put [here](https://github.com/thelounge/lounge/pull/986/files#diff-e6a5b42b2f7a26c840607370aed5301aR438)? In several cases, some of these attributes were not defined, so this was the best I could do to make it look nice.